core/src/test/scala/unit/kafka/server/AddPartitionsToTxnRequestServerTest.scala

```diff
@@ -55,22 +65,149 @@ class AddPartitionsToTxnRequestServerTest extends BaseRequestTest {
     val producerId = 1000L
     val producerEpoch: Short = 0
 
-    val request = new AddPartitionsToTxnRequest.Builder(
-      transactionalId,
-      producerId,
-      producerEpoch,
-      List(createdTopicPartition, nonExistentTopic).asJava)
-      .build()
+    val request =
+      if (version < 4) {
+        AddPartitionsToTxnRequest.Builder.forClient(
+          transactionalId,
+          producerId,
+          producerEpoch,
+          List(createdTopicPartition, nonExistentTopic).asJava)
+          .build()
```

Comment:
nit: `build()` looks weird here. Should we put the closing parenthesis of `forClient` on a new line: `).build()`?